File: codewars/py/frequency_sequence.py
from collections import Counter


def freq_seq(s, sep):
    count = Counter(s)
    res = []
    for letter in s:
        res.append(str(count[letter]))
    sequence = str(sep).join(res)
    return sequence

if __name__ == '__main__':
    s = 'hello world'
    sep = '-'
    print(freq_seq(s, sep))

File: codewars/js/frequency_sequence.js
function freqSeq(str, sep) {
    
    let res = ''
    let clean  = new Map()

    for (i=0; i< str.length; i++){
        if (str[i] && !clean[str[i]]){
            clean[str[i]] = 1
        }
        else{
            if (str[i]){
                clean[str[i]] += 1
            }
            
        }
    } 

    for (i = 0; i < str.length; i++){
        if (i == str.length -1){
            res += clean[str[i]]
        }
        else{
            res += (clean[str[i]] + sep)
        }
       
        
    }

    return res
}
